src/collections/Courses/index.ts

```diff
@@ -0,0 +1,216 @@
+import { populatePublishedAt } from '@/hooks/populatePublishedAt'
+import { TIMEZONE_OPTIONS } from '@/utilities/timezones'
+import { validateZipCode } from '@/utilities/validateZipCode'
+import { CollectionConfig } from 'payload'
+import { accessByProviderOrProviderManager } from './access/byProviderOrProviderManager'
+import { courseTypesData } from './constants'
+
+export const Courses: CollectionConfig = {
+  slug: 'courses',
+  access: accessByProviderOrProviderManager,
+  admin: {
+    group: 'Courses',
+    defaultColumns: ['title', 'subtitle', 'featuredImage', 'startDate', 'updatedAt'],
+    useAsTitle: 'title',
+  },
+  fields: [
+    {
+      name: 'title',
+      type: 'text',
+      required: true,
+    },
+    {
+      name: 'subtitle',
+      type: 'text',
+      admin: {
+        description: 'Optional subtitle/tagline for the event',
+      },
+    },
+    {
+      name: 'description',
+      type: 'textarea',
+      admin: {
+        description: 'Short description/summary for event previews',
+      },
+    },
+    startAndEndDateField(),
+    {
+      name: 'timezone',
+      type: 'select',
+      options: TIMEZONE_OPTIONS,
+      admin: {
+        description: 'Event timezone',
+      },
+    },
+    {
+      name: 'location',
+      type: 'group',
+      fields: [
+        {
+          name: 'placeName',
+          type: 'text',
+          label: 'Place Name',
+          required: true,
+        },
+        {
+          name: 'address',
+          type: 'text',
+          label: 'Street Address',
+        },
+        {
+          type: 'row',
+          fields: [
+            {
+              name: 'city',
+              type: 'text',
+              label: 'City',
+            },
+            {
+              name: 'state',
+              type: 'select',
+              label: 'State',
+              options: stateOptions,
+              required: true,
+            },
+            {
+              name: 'zip',
+              type: 'text',
+              label: 'ZIP Code',
+              validate: validateZipCode,
+            },
+          ],
+        },
+        {
+          name: 'country',
+          type: 'select',
+          options: [{ label: 'United States - US', value: 'US' }],
+          label: 'Country',
+          defaultValue: 'US',
+          admin: {
+            readOnly: true,
+          },
+        },
+        ...coordinatesWithMap({
+          condition: (_data, siblingData) => !siblingData?.isVirtual,
+        }),
+      ],
+    },
+    {
+      name: 'courseUrl',
+      type: 'text',
+      admin: {
+        description: 'External registration link or landing page link',
+      },
+    },
+    {
+      name: 'registrationDeadline',
+      type: 'date',
+      admin: {
+        date: {
+          pickerAppearance: 'dayAndTime',
+        },
+        description: 'Registration cutoff',
+      },
+      validate: (value, { siblingData }) => {
```

Comment:
:pick: I think you can avoid the `as` here by defining typing `siblingData: LinkFields`